File: hw/chavez/u3/hw21AnimalFarm/javascript/model/Cow.js
import FarmAnimal from "./FarmAnimal.js";
import Cut from "./Cut.js";
import SlaughterHouse from "./SlaughterHouse.js";

export default class Cow extends FarmAnimal {

    cut() {
        console.log("Cow has been cut");

        const cuts = [
            new Cut("Brisket"),
            new Cut("Rib"),
            new Cut("Loin"),
            new Cut("Chuck")
        ];

        cuts.forEach(cut => console.log(cut.toString()));
    }

    sendToSlaughterHouse(name) {
        const slaughterHouse = new SlaughterHouse(name);
        slaughterHouse.slaughter(this);
    }
}

File: hw/chavez/u3/hw21AnimalFarm/python/model/Cow.py
from model.FarmAnimal import FarmAnimal
from model.Cut import Cut
from model.SlaughterHouse import SlaughterHouse

class Cow(FarmAnimal):

    def cut(self):
        print("Cow has been cut")

        cuts = [
            Cut("Brisket"),
            Cut("Rib"),
            Cut("Loin"),
            Cut("Chuck")
        ]

        for cut in cuts:
            print(cut)

    def send_to_slaughter_house(self, name):
        slaughter_house = SlaughterHouse(name)
        slaughter_house.slaughter(self)